packages/typeahead/src/react/index.js

```diff
@@ -0,0 +1,257 @@
+import React, { Children } from 'react'
+import PropTypes from 'prop-types'
+
+import ActionMenu from '@pluralsight/ps-design-system-actionmenu/react.js'
+import core from '@pluralsight/ps-design-system-core'
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import Icon from '@pluralsight/ps-design-system-icon/react.js'
+import { BelowLeft } from '@pluralsight/ps-design-system-position/react.js'
+import { elementOfType } from '@pluralsight/ps-design-system-prop-types'
+import TextInput from '@pluralsight/ps-design-system-textinput/react.js'
+
+import * as vars from '../vars/index.js'
+
+import { combineFns, omit, pick } from './utils.js'
+
+const TEXT_INPUT_PROPS = [
+  'appearance',
+  'disabled',
+  'error',
+  'label',
+  'name',
+  'onChange',
+  'placeholder',
+  'subLabel',
+  'value'
+]
+const Typeahead = React.forwardRef((props, forwardedRef) => {
+  const { children, filterFn, onChange, value } = props
+
+  const portal = usePortal()
+  const containerRef = React.useRef()
+
+  const fieldRef = React.useRef()
+  const inputRef = React.useRef()
+  React.useImperativeHandle(forwardedRef, () => inputRef.current)
+
+  const [controlled] = React.useState(typeof value !== 'undefined')
+  const [open, setOpen] = React.useState(false)
+
+  const [innerValue, setInnerValue] = React.useState(value || '')
+  const [searchTerm, setSearchTerm] = React.useState('')
+
+  React.useEffect(() => {
+    if (controlled) setInnerValue(value)
+  }, [controlled, value])
+
+  React.useEffect(() => {
+    if (!open) setSearchTerm('')
+  }, [open])
+
+  useOnDocumentClick(evt => {
+    const isInnerClick =
+      !containerRef.current ||
+      containerRef.current.contains(evt.target) ||
+      portal.current.contains(evt.target)
+
+    if (isInnerClick) return
+
+    setOpen(false)
+  })
+
+  const suggestions = React.useMemo(() => {
+    const childArray = Children.toArray(children)
+    return childArray.map((child, index) => ({
+      comp: child,
+      index,
+      value: getSuggestionValue(child)
```

Comment:
As we discussed, I think there's an opportunity to:

1) Support entity selection
2) Be consistent with the Dropdown API

By adding `value` as a prop to `Typeahead.Suggestion`.